src/core/src/op/if.cpp

```diff
@@ -33,15 +33,21 @@ static ov::PartialShape resolve_shape(const ov::PartialShape& then_pshape, const
         return ov::PartialShape::dynamic();
     }
     if (then_rank.get_length() != else_rank.get_length()) {
+        auto isScalar = [](const ov::PartialShape& pshape) {
```

Comment:
This lambda is not required as in this blocks ranks of shapes  `then`, `else` are known and check their lengths should be sufficient.
Add test to validate new scenario, please.